how-to/hints-and-tips/docs/appassets-and-launch-external-process.md

```diff
@@ -0,0 +1,279 @@
+> **_:information_source: OpenFin Workspace:_** [OpenFin Workspace](https://www.openfin.co/workspace/) is a commercial product and this repo is for evaluation purposes (See [LICENSE.MD](../LICENSE.MD)). Use of the OpenFin Container and OpenFin Workspace components is only granted pursuant to a license from OpenFin (see [manifest](../public/manifest.fin.json)). Please [**contact us**](https://www.openfin.co/workspace/poc/) if you would like to request a developer evaluation key or to discuss a production license.
+
+[<- Back to Table Of Contents](../README.md)
+
+# App Assets and Launch External Process
+
+This hints and tips entry aims to cover these two features and how they can be used together. First we will cover a high level overview, then we will cover some common use cases and finally we take a deeper dive into these features followed by an example you can look at.
+
+## An overview of App Assets and Launch External Process
+
+```mermaid
+graph TD
+  subgraph Server
+    index.html
+    manifest.json
+    exe.zip
+  end
+  subgraph PC
+  Desktop.Shortcut -->|1. User clicks Shortcut| RVM
+    OpenFin.Application
+    AppData/Local/OpenFin/apps/YOUR_APP/assets/myApp/4.12.8
+    RVM -->|3. RVM reads appAsset and downloads exe.zip| AppData/Local/OpenFin/apps/YOUR_APP/assets/myApp/4.12.8
+    RVM --> |4. RVM reads runtime version, starting url and launches|OpenFin.Application
+    manifest.json -->|2. RVM reads manifest via HTTPS| RVM
+    exe.zip -->|HTTPS| RVM
+    index.html -->|5. Runtime loads url via HTTPS and executes JavaScript| OpenFin.Application
+    OpenFin.Application -->|6. Launch External Process launches myApp.exe| AppData/Local/OpenFin/apps/YOUR_APP/assets/myApp/4.12.8
+  end
+```
+
+In the example above we have a developer who has created a simple OpenFin Application. They have a .NET app that they wish to use with the OpenFin application. They have zipped up the app as exe.zip and placed it on their server. They have created a manifest.json file to specify the app asset they need, the version of OpenFin they wish to use, the permissions required and the url to load when launched (which launches the .NET app). The index.html page and manifest are uploaded to the server alongside the exe.zip.
+
+An end user installs the OpenFin Application and clicks on the desktop shortcut. The OpenFin installation has added the OpenFin RVM to the user's desktop. The desktop shortcut points to the RVM and passes the url to the manifest hosted on the developer's server. The RVM notes that the manifest lists an app asset and that it doesn't exist on disk. The RVM downloads and extracts the app asset into a versioned folder under the OpenFin Application's directory.
```

Comment:
Has it added it to the desktop, that's a bit confusing ?
`The OpenFin installation has added the OpenFin RVM to the user's desktop`

Reply:
The rvm is the installer and so on a clean machine the installer would add the rvm to the machine.

Reply:
But its not adding it to the desktop, its adding it to the machine - if it says its adding it to desktop I would expect something to appear on the desktop. `The OpenFin installation has added the OpenFin RVM to the user's machine`

Reply:
changed to machine.